hadoop-common-project/hadoop-common/src/main/java/org/apache/hadoop/ipc/Client.java

```diff
@@ -736,19 +733,17 @@ public Object run() throws IOException, InterruptedException {
                   + UserGroupInformation.getLoginUser().getUserName() + " to "
                   + remoteId;
               LOG.warn(msg, ex);
-              throw (IOException) new IOException(msg).initCause(ex);
+              throw new IOException(msg, ex);
```

Comment:
what about using NetUtils.wrapException which is how we should be reporting this stuff -it includes port info too where specified, and includes links to the wiki pages to help people debug it

Reply:
Destination address host:port would be included with this exception. However, yes `NetUtils.wrapException` is excellent utility and worth using here as it also covers wider variety of exceptions and updates the error msg accordingly.